src/doc/guide.md

```diff
@@ -408,6 +411,66 @@ will not fail your overall build. Please see the [Travis CI Rust
 documentation](https://docs.travis-ci.com/user/languages/rust/) for more
 information.
 
+# Templates
+
+Cargo uses the handlebars library to compile the templates used to scaffold
+projects. By default, there are only two templates available, `bin` and `lib`.
+These are used by cargo to create the standard project structure.
+
+You can also specify other templates from which to scaffold your project. The
+`--template` argument to `cargo new` accepts either a template name that you
+have downloaded on your system, or a URL to a remote Git repository containing
+the project template.
+
+```
+# use the mytemplate template which is located in ~/.cargo/templates/mytemplate
+$ cargo new myproj --template mytemplate
+
+# download the template called mytemplate from your github package
+$ cargo new myproj --template http://github.com/you/mytemplate
+```
+
+## Creating new templates
+
+A cargo template is just a folder containing one or more files. Usually, there
+is a `Cargo.toml` and a `src` directory. Each file in the template directory
+will be treated as a handlebars template. This means you can use handlebars
+variables wherever you want dynamic content, and cargo will render the proper
+values. Let's create a simple example. Create a new folder called `mytemplate`
+in your `~/.cargo/templates` directory..
+Add the following files:
+
+```toml
+# Cargo.toml
+[project]
+name = "{{name}}"
+version = "0.1.0"
+authors = [{{{authors}}}]
```

Comment:
This should use `toml-escape` now, right? Perhaps those various functions could be mentioned here as well?